apps/wakunode2/app.nim

```diff
@@ -512,6 +512,46 @@ proc setupProtocols(node: WakuNode, conf: WakuNodeConf,
       pubsubTopics = conf.topicsDeprecated.split(" ")
     else:
       pubsubTopics = conf.topics
+
+    # Validate topics and update a node's ENR accordingly
+    let parsed_topics = pubsubTopics.mapIt(NsPubsubTopic.parse(it))
+
+    var prevTopic: NsPubsubTopic
+    for i, topic in parsed_topics:
+      if topic.isErr():
+        return err("Pubsub topic parsing error")
+      
+      if i == 0:
+        prevTopic = topic.value
+        continue
+      
+      if prevTopic.kind != topic.value.kind:
+        return err("Please use named topics OR sharded ones not both.")
+        
+      if prevTopic.kind == NsPubsubTopicKind.StaticSharding:
+        if prevTopic.shard != topic.value.shard:
+          return err("Please use sharded topics within the same shard.")
+
+      if i > 0:
+        prevTopic = topic.value
+
+    if prevTopic.kind == NsPubsubTopicKind.StaticSharding:
+      let indices = parsed_topics.mapIt(it.value.shard)
+      let relayShard = RelayShards.init(prevTopic.cluster, indices)
+    
+      var record = node.wakuDiscv5.protocol.localNode.record
+      let key = node.wakuDiscv5.protocol.privKey()
+      
+      let pairs = if (let idxList = relayShard.toIndicesList(); idxList.isOk):
+        [toFieldPair(ShardingIndicesListEnrField, idxList.value)]
+      else:
+        let bitVector = relayShard.toBitVector()
+
+        [toFieldPair(ShardingBitVectorEnrField, bitVector)]
+
+      if record.update(key, pairs).isErr():
```

Comment:
wondering if this logic should go in `initNode` or `WakuDiscoveryV5.new`.

`flags = netConfig.wakuFlags.get()` is similar, see [this](https://rfc.vac.dev/spec/31/#waku2-enr-key), so perhaps its a good idea to do all enr field modification in the same place: so far it would be shards + wakuFlags.

Reply:
The discv5 re-org changes a lot of thing. Notably where the ENR will be created. IDK where it fit yet.